airflow/contrib/operators/databricks_operator.py

```diff
@@ -0,0 +1,189 @@
+        endpoint.
+    :type extra_api_parameters: dict
+    :param databricks_conn_id: The name of the connection to use.
+        By default and in the common case this will be ``databricks_default``.
+    :type databricks_conn_id: string
+    """
+    ui_color = '#1CB1C2'
+    ui_fgcolor = '#fff'
+
+    def __init__(
+            self,
+            spark_jar_task=None,
+            notebook_task=None,
+            new_cluster=None,
+            existing_cluster_id=None,
+            libraries=[],
+            run_name=None,
+            timeout_seconds=0,
+            extra_api_parameters={},
+            databricks_conn_id='databricks_default',
+            **kwargs):
+        """
+        Creates a new ``DatabricksSubmitRunOperator``.
+        """
+        super(DatabricksSubmitRunOperator, self).__init__(**kwargs)
+        self.spark_jar_task = spark_jar_task
+        self.notebook_task = notebook_task
+        self.new_cluster = new_cluster
+        self.existing_cluster_id = existing_cluster_id
+        self.libraries = libraries
+        self.run_name = kwargs['task_id'] if run_name is None else run_name
+        self.timeout_seconds = timeout_seconds
+        self.extra_api_parameters = extra_api_parameters
+        self.databricks_conn_id = databricks_conn_id
+        self._validate_parameters()
+
+    def _validate_parameters(self):
+        """
+        Validates the parameters provided to this operator.
+        """
+        self._validate_oneof(self.spark_jar_task, self.notebook_task)
+        self._validate_oneof(self.new_cluster, self.existing_cluster_id)
+
+    def _validate_oneof(self, param_a, param_b):
+        """
+        Ensures either param_a XOR param_b is set.
+        """
+        if param_a is not None and param_b is None:
+            pass
+        elif param_a is None and param_b is not None:
+            pass
+        else:
+            raise AirflowException(('It is an error to provide both {0} and {1}. ' +
+                'Please select only one').format(param_a, param_b))
+
+    def _log_run_page_url(self, url):
+        logging.info('View run status, Spark UI, and logs at {}'.format(url))
+
+    def get_hook(self):
+        return DatabricksHook(self.databricks_conn_id)
+
+    def execute(self, context):
+        hook = self.get_hook()
+        run_id = hook.submit_run(
+            self.spark_jar_task,
+            self.notebook_task,
+            self.new_cluster,
+            self.existing_cluster_id,
+            self.libraries,
+            self.run_name,
+            self.timeout_seconds,
+            **self.extra_api_parameters)
+        run_page_url = hook.get_run_page_url(run_id)
+        logging.info(LINE_BREAK)
+        logging.info('Run submitted with run_id: {}'.format(run_id))
+        self._log_run_page_url(run_page_url)
+        logging.info(LINE_BREAK)
+        while True:
+            run_state = hook.get_run_state(run_id)
+            if run_state.is_terminal:
```

Comment:
This would run into infinite loops if we introduce a new terminal state and `is_terminal` cannot recognize it.